src/Domain/LeanCode.CQRS.Validation.Fluent.Scoped/AdapterLoader.cs

```diff
@@ -0,0 +1,29 @@
+using System.Threading.Tasks;
+using Autofac;
+using FluentValidation;
+using LeanCode.Contracts;
+using LeanCode.Contracts.Validation;
+
+namespace LeanCode.CQRS.Validation.Fluent.Scoped;
+
+internal class AdapterLoader<TAppContext, TCommand> : ICommandValidator<TAppContext, TCommand>
+    where TAppContext : notnull
+    where TCommand : ICommand
+{
+    private static readonly Task<ValidationResult> NoError = Task.FromResult(new ValidationResult(null));
+
+    private readonly FluentValidationCommandValidatorAdapter<TAppContext, TCommand>? adapter;
+
+    public AdapterLoader(IComponentContext ctx)
+    {
+        if (ctx.TryResolve<IValidator<TCommand>>(out var val))
```

Comment:
We can't resolve the validator up to a point where we know the `appContext` value - it can't be here.

Reply:
I may be wrong, but I don't think that the registered validator takes an appContext as a parameter. So I don't understand why we need appContext to resolve it?

Reply:
It related to [this comment](https://github.com/leancodepl/corelibrary/pull/465/files#r1118740477) - it _has to_ takee it if you want to use it, because you won't have other way to get to it.